### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by enoqueJonas**
# OOP school library: decorate a class

## This PR was raised to:
- Implement composition (as an example of the design pattern).
- Run a program using the command line.

## What does this PR do:
- Create a class Nameable.
- Turn Person class to Nameable.
- Prepare base Decorator.
- Prepare CapitalizeDecorator and TrimmerDecorator

## Improvements to be made:
- Add a menu to run the app

### CODE REVIEW

File: decorators/nameable_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Nameable
+  def correct_name; end
```
Comment (franclobo) [inline decorators/nameable_decorator.rb]: - [ ] This is a requirement to implement a method called correct_name that will ** raise a NotImplementedError**. Please 🙏🏽  complete the correct_name method to raise a NotImplementedError. We could review this [link](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0) for help.

File: decorators/captilized_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require_relative './base_decorator'
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (franclobo) [inline decorators/captilized_decorator.rb]: - [ ] `Capitalize` and `Upcase` are different methods as we need to `capitalize` please 🙏🏽 use the correct method for your CapitalizeDecorator class work as expected.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @enoqueJonas, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 👏🏽 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good use of inheritance 💯 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**